microcosm_pubsub/fields.py

```diff
@@ -0,0 +1,45 @@
+"""
+Custom fields.
+"""
+from marshmallow.fields import Field, ValidationError
+from uuid import UUID
+
+
+class UUIDField(Field):
+    """
+    UUID valued field, as either a string or UUID object.
+
+    """
+    def __init__(self, use_uuidformat=False, *args, **kwargs):
+        self.use_uuidformat = use_uuidformat
+        super(UUIDField, self).__init__(*args, **kwargs)
+
+    def _serialize(self, value, attr, obj):
+        """
+        Serialize value as a uuid, either as a string or an UUID object.
+        """
+        if value is None:
+            return None
+
+        if self.use_uuidformat:
+            return str(value)
+        else:
+            return value
+
+    def _deserialize(self, value, attr, data):
+        """
+        Deserialize value as a uuid.
+
+        Handle both string and UUID object.
+        """
+        if value is None:
+            return None
+
+        try:
+            if self.use_uuidformat:
```

Comment:
You don't have control of your inputs in all cases. It's equally plausible that you are passed a `UUID` or a string. But no matter what you are passed, we want to return a string so that the pubsub layer doesn't blow up. Right?